tox.ini

```diff
@@ -13,55 +13,40 @@ envlist =
 requires =
     setuptools >= 30.3.0
     pip >= 19.3.1
-    tox-pypi-filter >= 0.12
-
-[testenv]
-pypi_filter = https://raw.githubusercontent.com/astropy/ci-helpers/main/pip_pinnings.txt
 
 extras =
     test
-    all
-
-commands =
-    pip freeze
-    pytest --pyargs pyvo
+    alldeps: all
 
-# tox environments are constructed with so-called 'factors' (or terms)
-# separated by hyphens, e.g. test-devdeps-cov. Lines below starting with factor:
-# will only take effect if that factor is included in the environment name. To
-# see a list of example environments that can be run, along with a description,
-# run:
-#
-#     tox -l -v
-#
 description =
     run tests
-    oldestdeps: with astropy 4.1.*
-    devastropy: with astropy latest
+    oldestdeps: with oldest supported dependencies
+    devdeps: with development version of dependencies
+    cov: determine the code coverage
 
 deps =
-    devastropy: git+https://github.com/astropy/astropy.git#egg=astropy
+    cov: coverage
+
     oldestdeps: astropy==4.1
     # We set a suitably old numpy along with an old astropy, no need to pick up
     # deprecations and errors due to their unmatching versions
     oldestdeps: numpy==1.16
 
-[testenv:egg_info]
-description = ensure egg_info works without dependencies
 commands =
+    devdeps: pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+    devdeps: pip install -U --pre -i https://pypi.anaconda.org/astropy/simple astropy
+
     pip freeze
-    python setup.py egg_info
+    !cov: pytest --pyargs pyvo
+    cov: pytest --pyargs pyvo --cov pyvo --cov-config={toxinidir}/setup.cfg
+    cov: coverage xml -o {toxinidir}/coverage.xml
 
-[testenv:cov]
-description = determine the code coverage
-deps:
-     # had to pin this
-    coverage==4.5.4
 
+[testenv:egg_info]
+description = ensure egg_info works without dependencies
 commands =
     pip freeze
-    pytest --pyargs pyvo --cov pyvo --cov-config={toxinidir}/setup.cfg
-    coverage xml -o {toxinidir}/coverage.xml
+    python setup.py egg_info
```

Comment:
Is this still needed? I think astropy stopped testing this a long time ago.

Reply:
yes, I suppose it can go.